Dockerfile

```diff
@@ -8,54 +9,79 @@ RUN cd llama.cpp && \
     make && \
     mv main llama
 
-# Copy over rest of the project files
+# Base image for node
+FROM node:19 as node_base
 
-FROM ubuntu:22.04 as deployment
+WORKDIR /usr/src/app
+# Install pip and requirements
+
+# Base image for runtime
+FROM ubuntu:22.04 as base
 
 ARG DEBIAN_FRONTEND=noninteractive
 ENV TZ=Europe/Amsterdam
 
 WORKDIR /usr/src/app
 
+# Install MongoDB and necessary tools
+RUN apt update && \
+    apt install -y curl wget gnupg && \
+    wget -qO - https://www.mongodb.org/static/pgp/server-6.0.asc | apt-key add - && \
+    echo "deb [ arch=amd64,arm64 ] https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/6.0 multiverse" | tee /etc/apt/sources.list.d/mongodb-org-6.0.list && \
+    apt-get update && \
+    apt-get install -y mongodb-org
+
+
 # install pip
-RUN apt update
-RUN apt-get install -y python3-pip curl wget
+RUN apt update && \
+    apt install -y python3-pip
 RUN pip install --upgrade pip
```

Comment:
This line can be merge with the RUN command above where mongodb-org and python3-pip get installed